src/compiler/checker.ts

```diff
@@ -11707,7 +11707,7 @@ module ts {
                 return getEnumMemberValue(<EnumMember>node);
             }
 
-            let symbol = getNodeLinks(node).resolvedSymbol;
+            let symbol = getSymbolAtLocation(node);

```

Comment:
Would it be easier to split `getConstantValue` into two functions: an internal and an external one? The external function can then call `getSymbolAtLocation`